class/practice2.1/AdvancedHelloWorld.cpp

```diff
@@ -1,110 +1,94 @@
+        if (statsData.find(nameToFind)!=statsData.end())
+        {
+            statsData [nameToFind]++;
+            returnValue="Hello again(x" + std::to_string(statsData[nameToFind]) + ")," + nameToFind;
+        }else{
+            statsData [nameToFind]++;
+            returnValue="Welcome " +nameToFind;
         }
+        SaveData();
+        return returnValue;
     }
-    if (!found) {
-        if (argument2 == "delete") {
-            std::cout << "There is no such user " <<std::endl;
-            return 0;
-        }
-        std::cout << "Welcome " << WordToFind << std::endl;
-        std::ofstream fileEdit("database.txt", std::ios::app);
-        if (!fileEdit.is_open()) {
-            std::cerr << "Could not open the file for writing!" << std::endl;
-            return 1;
-        }
-        fileEdit << WordToFind << " 1" << std::endl;
-        fileEdit.close();
+public:
+    Database(const std::string& filePath) : database(filePath) {
+         LoadData();
     }
-    else {
-        if (argument2 == "delete") {
-
-            for (int i = 0; i < lines.size(); i++)
-            {
-                if (splitString(lines[i], ' ')[0] == WordToFind)
-                {
-                    lines.erase(lines.begin() + i);
-                }
+    void clearDatabase() {
+        statsData.clear();
+        SaveData();
+    }
+    std::string PerformChanges(const std::string& nameToFind, const std::string& additionalCommand) {
+        if (statsData.find(nameToFind)==statsData.end()) {
+            if (additionalCommand == deleteCommand) {
+               return "There is no such user ";
             }
-            std::ofstream fileEdit("database.txt", std::ios::trunc);
-            for (const auto& line : lines) {
-                if (line != "")
-                {
-                    fileEdit << line << std::endl;
-                }
+            return UserIncrement(nameToFind);
+        }
+        else {
+            if (additionalCommand == deleteCommand) {
+                return deleteUser(nameToFind);
             }
-            std::cout << "All data about this user was cleared" << std::endl;
-            return 0;
+            return UserIncrement(nameToFind);
         }
-
-        std::ofstream fileEdit("database.txt", std::ios::trunc);
-        std::vector<std::string> arraySplitted=splitString(lineFound, ' ');
-       int num = std::stoi(arraySplitted[1]);
-       num++;
-       lines[lineNumber - 1] = arraySplitted[0] +" " + std::to_string(num);
-       for (const auto& line : lines) {
-           if (line != "")
-           {
-               fileEdit << line << std::endl;
-           }
-       }
-       std::cout << "Hello again(x"<<num <<"),"<< WordToFind << std::endl;
-       fileEdit.close();
     }
+};
+int main(int argc, char* argv[])
+{
+    Database db("database.txt");
+    if (argc < 2) {
+        std::cout << "Please provide a name to find." << std::endl;
+        return 0;
+    }
+    std::string nameToFind;
```

Comment:
add empty lines between logical blocks - it improves readability.
also create database after handling of arguments. also return 1 as error should not return 0, use println if possible:
```
if (argc < 2) {
     std::println(...);
     return 1;
}

Database db("...");
```
use named constant for database filepath